mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -217,15 +217,17 @@ void mlir::tosa::printTypeOrAttr(OpAsmPrinter &p, Operation *op, TypeAttr type,
 
 template <typename T>
 static LogicalResult verifyConvOp(T op) {
-  // All TOSA conv ops have an input and weight arguments which must be ranked
-  // tensors.
+  // All TOSA conv ops have an input() and weight().
   auto inputType = llvm::dyn_cast<RankedTensorType>(op.getInput().getType());
+
+  RankedTensorType weightType;
+  weightType = llvm::dyn_cast<RankedTensorType>(op.getWeight().getType());
```

Comment:
Why move it up and not declare it before use?

Reply:
I don't think there's any particular reason. We can move it down if you prefer that way.

Reply:
removed